File: example/env/flutter/material/appBar.ts
import { Widget } from './../widget';
import { SyntheticBox } from '../../syntheticBox';

interface AppBarProps {
    title: Widget;
}

declare const flutter: {
    material: {
        appBar: (this: void, props: AppBarProps) => { tag: string };
    }
}

export class AppBar extends SyntheticBox {
    public props: AppBarProps;
    public constructor(props: AppBarProps) {
        super();
        this.props = props;
    }

    public unwrap(): Widget {
        return flutter.material.appBar(this.props);
    }
}

File: lib/builtins/flutter/material/appBar.dart
import 'package:flua/hydroState.dart';
import 'package:flua/vm/context.dart';
import 'package:flua/builtins/flutter/syntheticBox.dart';
import 'package:flua/vm/table.dart';
import 'package:flutter/material.dart';

loadAppBar({@required HydroState luaState, @required HydroTable table}) {
  table["appBar"] = makeLuaDartFunc(func: (List<dynamic> args) {
    return [
      AppBar(
        title: maybeUnwrapAndBuildArgument<Widget>(args[0]["title"],
            parentState: luaState),
        leading: maybeUnwrapAndBuildArgument<Widget>(args[0]["leading"],
            parentState: luaState),
        actions: maybeUnwrapAndBuildArgument<Widget>(args[0]["actions"],
            parentState: luaState),
        bottom: maybeUnwrapAndBuildArgument<Widget>(args[0]["bottom"],
            parentState: luaState),
      )
    ];
  });
}
